runtime/v2/example/example.go

```diff
@@ -0,0 +1,145 @@
+// +build linux
+
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package example
+
+import (
+	"context"
+	"os"
+	"time"
+
+	"github.com/containerd/containerd/events"
+	"github.com/containerd/containerd/runtime/v2/shim"
+	taskAPI "github.com/containerd/containerd/runtime/v2/task"
+	ptypes "github.com/gogo/protobuf/types"
+	"golang.org/x/sys/unix"
+)
+
+var (
+	// check to make sure the *service implements the GRPC API
+	_ = (taskAPI.TaskService)(&service{})
+	// common response type
+	empty = &ptypes.Empty{}
+)
+
+// New returns a new shim service
+func New(ctx context.Context, id string, publisher events.Publisher) (shim.Shim, error) {
+	return &service{}, nil
+}
+
+type service struct {
+}
+
+// StartShim is a binary call that executes a new shim returning the address
+func (s *service) StartShim(ctx context.Context, id, containerdBinary, containerdAddress string) (string, error) {
+	return "", nil
+}
+
+// Cleanup is a binary call that cleans up any resources used by the shim when the service crashes
+func (s *service) Cleanup(ctx context.Context) (*taskAPI.DeleteResponse, error) {
+	return &taskAPI.DeleteResponse{
+		ExitedAt:   time.Now(),
+		ExitStatus: 128 + uint32(unix.SIGKILL),
+	}, nil
+}
+
+// Create a new container
+func (s *service) Create(ctx context.Context, r *taskAPI.CreateTaskRequest) (_ *taskAPI.CreateTaskResponse, err error) {
+	return &taskAPI.CreateTaskResponse{}, nil
```

Comment:
This violates the spec :). You should `return nil, errdefs.ErrNotImplemented` on all these